### PR TITLE
Sils/formakema

### PR DESCRIPTION
**Pull request by Makman2**
The desired windows fix

### CODE REVIEW

**Comment by Makman2** [comment]
Issues to file after merge:
- Replace `_to_list` with a public API from coala-decorators (which needs to be implemented there too right)

**Comment by sils1297** [comment]
tests fail, marking WIP.

**Comment by Makman2** [comment]
Excluded the "nice" fix for the bear importing problem, as it's not ready yet, though there are other little nice commits that should be merged :)

File: coalib/collecting/Importers.py [hunk 1]
```diff
@@ -146,15 +147,12 @@ def iimport_objects(file_paths, names=None, types=None, supers=None,
...
-            for obj in _iimport_objects(file_paths, names, types, supers,
-                                        attributes, local):
-                yield obj
-    else:
-        for obj in _iimport_objects(file_paths, names, types, supers,
-                                    attributes, local):
-            yield obj
+    with ExitStack() as stack:
+        if not verbose:
+            stack.enter_context(suppress_stdout())
```
Comment (Makman2) [inline coalib/collecting/Importers.py]: do we actually need this? :O

File: coalib/collecting/Collectors.py [hunk 2]
```diff
@@ -11,6 +10,8 @@
 from coalib.parsing.Globbing import fnmatch, iglob, glob_escape
 from coalib.output.printers.LogPrinter import LogPrinter
 
+from pyprint.NullPrinter import NullPrinter
```
Comment (Makman2) [inline coalib/collecting/Collectors.py]: oh I think this needs to come before `coalib`, right?

File: coalib/collecting/Importers.py (hunk 1 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: ahm isn't this your own commit?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/98ea966b9ee61e4e77959f23fadbad95fa55bab7

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/0f5b414a02b32d1f0dbd823ff6862a027edfc7c0 where does that one come from!?

File: coalib/collecting/Importers.py [hunk 3]
```diff
@@ -143,6 +143,8 @@ def iimport_objects(file_paths, names=None, types=None, supers=None,
     :param attributes: Attribute(s) an object needs to (all) have
     :param local:      if True: Objects need to be defined in the file they
                        appear in to be collected
+    :param verbose:    Whether verbose logs shall be displayed on console or
+                       not.
```
Comment (sils1297) [inline coalib/collecting/Importers.py]: nope, this is not about logs, this is about suppression of STDOUT!

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/c2a2fc18f7ada72696de9d5e52c800752cb275cb

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/81aa63dea46bf4b5c7948715b36c1e5a0d12dd4e

File: coalib/collecting/Importers.py [hunk 4]
```diff
@@ -113,20 +113,16 @@ def _iimport_objects(file_paths, names, types, supers, attributes, local):
     :raises Exception: Any exception that is thrown in module code or an
                        ImportError if paths are erroneous.
     """
-    if (file_paths == [] or
-            (names == [] and
-             types == [] and
-             supers == [] and
-             attributes == [])):
-        raise StopIteration
+    if not (file_paths and (names or types or supers or attributes)):
```
Comment (sils1297) [inline coalib/collecting/Importers.py]: :D why negating the condition? I think the other form was easier to understand

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/0aa3c765c16acb04cf800ebf85da825f2a31a899

**Comment by sils1297** [comment]
don't trust your performance measurements yet but I guess I'll ack the same commit next time I read it :P

File: coalib/collecting/Importers.py (hunk 3 above)
Comment (Makman2) [inline coalib/collecting/Importers.py]: okay okay keep cool, though then the name "verbose" is kinda wrong^^ but do we need "verbose" actually?

File: coalib/collecting/Importers.py (hunk 4 above)
Comment (Makman2) [inline coalib/collecting/Importers.py]: I double-negated it, and expanded one negation onto the whole expression^^
if you think I should keep the other version, then we should at least do
`if (not file_paths or (not names and not ...))`

**Comment by Makman2** [comment]
why you don't trust the performance measurements? I mean I just did them :O

File: coalib/collecting/Importers.py (hunk 1 above)
Comment (Makman2) [inline coalib/collecting/Importers.py]: I mean the verbose parameter^^

**Comment by sils1297** [comment]
:P

File: coalib/collecting/Importers.py (hunk 3 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: I'm totally cool! No I'm not actually :P everybody keeps pinging me. Can you take over reviewing stuff for @Redridge and @mr-karan ? They need their stuff merged as today is the deadline for that...

**Comment by Makman2** [comment]
reack
6f3c4e1 Processing: Be PEP8 conform
e610649 Importers: Remove code duplication

File: coalib/collecting/Importers.py (hunk 3 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: @Makman2 this commit still adds wrong docs

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/866413d86515e748e59c425e7a2baffee6d038d4

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/eed402712014c7a22886d798597c237f4da3bd07

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/0a1fed49917f98b7e5958177e409c506377bc928 https://github.com/coala-analyzer/coala/pull/2345/commits/4192b975161bb9f63981150007bd97790d0e605c

File: coalib/collecting/Importers.py (hunk 4 above)
Comment (sils1297) [inline coalib/collecting/Importers.py]: not so sure here, comparison with empty is not the same as `not bool()` of an object

Comment (sils1297) [inline coalib/collecting/Importers.py]: this has breakage potential for corner cases and it's easier to understand in the original version IMO. UNack from me.

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/79f1fba8651a9b52b3437ff5bbafe11b582baa07

File: coalib/collecting/Importers.py (hunk 3 above)
Comment (Makman2) [inline coalib/collecting/Importers.py]: then give me some inputs, `verbose` means for me verbose logs are outputted.

File: coalib/collecting/Importers.py (hunk 4 above)
Comment (Makman2) [inline coalib/collecting/Importers.py]: in this case it is because every argument is made into a list, and `bool` on list is well defined^^

**Comment by Makman2** [comment]
ack
8e55ead MANIFEST.in: Point to README.rst and not to .md
8cd5c77 Future: Remove list_dir_contents
f044eb3 Collectors: Add newline between imports
5c5a59a Processing: Be PEP8 conform
0e7f001 Importers: Remove code duplication

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/ab5aaf1e78e83807a7fc686bef87bc510b066f20

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/2e046183d94a7f0731b08772d0291db00635d33b apparently you don't rename the parameter in the signature in that commit which means that some other commit of yours breaks our code!

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2345/commits/6a21bc6dce31d8e5dfd06ec88f85780cd8de10ce as it breaks the code by renaming the parameter but not it's usages

**Comment by Makman2** [comment]
reack
91a32dc MANIFEST.in: Point to README.rst and not to .md
e3bc74f Future: Remove list_dir_contents
15a6b99 Importer._iimport_objects: Simplify
b61f12a Collectors: Add newline between imports
ff2d5ec Processing: Be PEP8 conform
b50159e Importers: Remove code duplication

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/86538ff08715952dde0161fe6b015e540c21dd64

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2345/commits/075bba1abce29306e34d0b692b0de1d065aaadef

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**